### PR TITLE
OpenAPI docs improvements

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
See /swagger endpoint.

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2908 Migrate Swagger -> OpenAPI and revise docs if they make sense and make sure all "hidden" params are documented](https://app.clickup.com/t/2192114/QF-2908)

**Comment by suricactus** [comment]
Thanks for removing the hook trick, I think this one has less maintanance weight for the future.

BTW, can you also double check if all the endpoints are properly documented. This can be easily tested using the swagger UI sending requests. If there is a missing parameter, then the requests will fail. Sometimes there are hidden parameters, also have a look on them too.

File: docker-app/qfieldcloud/core/views/files_views.py [hunk 1]
```diff
@@ -125,6 +141,27 @@ def has_permission(self, request, view):
...
+    post=extend_schema(
+        description="Upload a file to the project",
+        parameters=[
+            OpenApiParameter(
+                "file",
+                OpenApiTypes.STR,
+                OpenApiParameter.QUERY,
+                required=True,
+                description="Path to the file to upload",
+            )
```
Comment (suricactus) [inline docker-app/qfieldcloud/core/views/files_views.py:160]:
```suggestion
            OpenApiParameter(
                "file",
                OpenApiTypes.BINARY,
                OpenApiParameter.QUERY,
                required=True,
                description="File to be uploaded",
            )
```

Comment (suricactus) [reply docker-app/qfieldcloud/core/views/files_views.py:160]: This is file, did you test actually making requests using the Swagger UI?

Comment (why-not-try-calmer) [reply docker-app/qfieldcloud/core/views/files_views.py]: It's on the menu -- once you confirm the API docs suits you :)

**Review by suricactus — COMMENTED** [review]
Code looks ok.

Something I forgot to mention is that /qfield-files endpoint is deprecated, so please hide it from the OpenAPI docs too.

The rest should be verified if requests from the Swagger UI work and fixes should be accomodated.

**Pull request merged**